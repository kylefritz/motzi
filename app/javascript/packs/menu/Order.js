import React from "react";
import _ from "lodash";

import BakersNote from "./BakersNote";
import Subscription from "./Subscription";
import Cart from "./Cart";

export default function Order({
  menu,
  user,
  order,
  onRefreshUser,
  onEditOrder,
}) {
  const { name, deadlineDay } = menu;
  const {
    items: cart,
    skip,
    comments,
    stripeReceiptUrl,
    stripeChargeAmount,
  } = order;

  const isSubscriptionOrder = _.get(user, "subscriber");
  return (
    <>
      <h2 id="menu-name">{name}</h2>

      <h3 className="mt-5 mb-4">We've got your order!</h3>

      <Cart {...{ cart, menu, skip, stripeChargeAmount }} />
      {stripeReceiptUrl && (
        <p className="text-center my-2">
          <a href={stripeReceiptUrl} target="blank">
            View receipt
          </a>
        </p>
      )}
      <div className="ml-2 mt-3">
        <h6>Feedback, Comments & Special Requests</h6>
        <p>{comments || <em>none</em>}</p>
      </div>

      {onEditOrder && isSubscriptionOrder && (
        <div>
          <button
            type="button"
            className="btn btn-outline-primary btn-sm"
            onClick={onEditOrder}
          >
            Edit Order
          </button>
        </div>
      )}

      {isSubscriptionOrder && (
        <div className="mt-5">
          <Subscription {...{ user, onRefreshUser, deadlineDay }} />
        </div>
      )}
    </>
  );
}
