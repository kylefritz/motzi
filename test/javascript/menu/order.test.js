import React from 'react';
require('../configure_enzyme');
import { shallow } from 'enzyme';

import Order from 'menu/Order'
import orderData from './order-data'

test('Order snapshot', () => {
  const wrapper = shallow(<Order {...orderData} />);
  expect(wrapper).toMatchSnapshot();
});
