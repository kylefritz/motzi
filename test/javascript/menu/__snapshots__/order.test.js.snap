// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order snapshot 1`] = `
<Fragment>
  <h2
    id="menu-name"
  >
    week 5
  </h2>
  <h3
    className="mt-5 mb-4"
  >
    We've got your order!
  </h3>
  <CartWrapper
    cart={
      Array [
        Object {
          "day": "Tuesday",
          "day1Pickup": true,
          "itemId": 3,
          "quantity": 1,
        },
        Object {
          "day": "Thursday",
          "day1Pickup": true,
          "itemId": 1,
          "quantity": 1,
        },
      ]
    }
    menu={
      Object {
        "createdAt": "2019-11-02T14:01:23.820-04:00",
        "day1": "Tuesday",
        "day1Deadline": "2019-11-03T23:59:59.000-04:00",
        "day2": "Thursday",
        "day2Deadline": "2019-11-05T23:59:59.000-04:00",
        "id": 921507399,
        "isCurrent": true,
        "items": Array [
          Object {
            "day1": true,
            "day2": true,
            "description": "",
            "id": 3,
            "image": "bread-baguette.jpg",
            "marketplace": true,
            "name": "Baguette",
            "price": 3,
            "subscriber": true,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "Mix of modern wheats and ancient Einkorn for the best of both worlds.",
            "id": 1,
            "image": "bread2-002.webp",
            "marketplace": true,
            "name": "Classic",
            "price": 4,
            "subscriber": true,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "ony subscribers can get cookies",
            "id": 2,
            "marketplace": false,
            "name": "Cookies",
            "price": 4,
            "subscriber": true,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "too small for subscribers",
            "id": 4,
            "marketplace": true,
            "name": "Marketplace only item",
            "price": 2,
            "subscriber": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "too small for subscribers",
            "id": 5,
            "marketplace": true,
            "name": "Another small item",
            "price": 1.5,
            "subscriber": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "Skip this week.",
            "id": 0,
            "name": "Skip",
            "subscriber": true,
          },
        ],
        "menuNote": "menu note copy",
        "name": "week 5",
        "subscriberNote": "subscribers note copy",
      }
    }
    skip={false}
  />
  <div
    className="ml-2 mt-3"
  >
    <h6>
      Feedback, Comments & Special Requests
    </h6>
    <p>
      <em>
        none
      </em>
    </p>
  </div>
</Fragment>
`;
